handlers/api/callback.go

```diff
@@ -75,39 +64,17 @@ func CallbackGet(c *fiber.Ctx) error {
 		return c.Next()
 	}
 
-	user, err := models.FindUserByNameOrEmail(response.Username, response.Email)
+	user, err := flow(response)
 	if err != nil {
-		if err.Error() != "User not found." && err.Error() != "record not found" {
-			return c.Next()
-		}
-		user = &models.User{
-			Username:      response.Username,
-			Email:         response.Email,
-			Role:          models.Regular,
-			OAuthProvider: service,
-		}
-		regErr := database.Conn.Create(user)
-
-		if regErr.Error != nil {
-			log.Warn.Printf("Failed to register %s: %s", response.Username, regErr.Error)
-			return c.Status(fiber.StatusInternalServerError).
-				JSON(fiber.Map{
-					"data": "Internal Error.",
-				})
-		}
-	}
-
-	// TODO: Simplify this logic.
-	if (user.OAuthProvider == "none" || user.OAuthProvider != service) &&
-		!strings.EqualFold(getSocialMediaValue(user, service), response.Username) {
-		log.Warn.Println("User detected but the social media value wasn't set of this user.")
-		return c.Next()
+		log.Warn.Printf("User %q failed to sign in: %s\n", response.Username, err)
+		msg := "Please contact us and provide this timestamp: " + time.Now().Format(time.RFC3339)
+		return c.Render("err", fiber.Map{"Title": msg})
 	}
 
 	expiration := time.Now().Add(time.Hour * 24 * 14)
 	t, err := utils.NewJWTToken().
 		SetClaim("id", user.ID).
-		SetClaim("name", user.Username).
+		SetClaim("name", strings.ToLower(user.Username)).
```

Comment:
It should be. I plan to remove it and switch to saving usernames as `COLLATE NOCASE` in the migration, so that we don't have to deal with needless edge-cases, such as this one. That would also allow users to keep capitalized letters in their usernames, which is something I didn't know how to do properly way back when I made them all lowercase.

Reply:
Alright, if the code can handle this, LGTM.